lib/Runtime/PlatformAgnostic/ICU.h

```diff
@@ -8,7 +8,15 @@
 
 #ifdef HAS_ICU
 #ifdef WINDOWS10_ICU
+// if WINDOWS10_ICU is defined, pretend like we are building for recent Redstone,
+// even if that isn't necessarily true
+#ifdef NTDDI_VERSION
+#undef NTDDI_VERSION
+#endif
+#define NTDDI_VERSION NTDDI_WIN10_RS3
 #include <icu.h>
```

Comment:
If you are limiting yourself to RS3+, then there is only 1 header file to include `icu.h`.

Reply:
Makes sense, see the comments above -- I am fairly certain that <icu.h> was resolving to the current file and failing silently. Shift-Control-G in VS doesnt bring me anywhere and you cant peek the file either. Ill change this to pretend to be RS2 and only use common and i18n

Reply:
In that case, would pretending to be RS2 allow it to work RS3+?

Reply:
Yup, forward-compat should always be a thing, I think. With that being said, after looking at this more I have decided to rename our own ICU.h to ChakraICU.h since it seems like <icu.h> is the preferred header going forward. Ill make an update to this today.

Reply:
@jackhorton Right, they should be **mostly** forward compatible.

However, one of the main reasons for creating a new header file (icu.h) was also due to the breaking-change that ICU made in version 59 of having the `UChar` typedef now use the new type `char16_t` everywhere on all platforms -- whereas before it was using platform-specific types, which happened to be `wchar_t` on Windows.

Since we only expose C APIs in the Kit version, and since the types `char16_t` and `wchar_t` are binary compatible (but not source compatible), there is fortunately no binary breaking change.

However, they may be some source breaking changes if you have code that makes use of `UChar`s.

For example, the following snippet will work with the RS2 headers, but not the RS3 headers:
  `UChar test1[] = L"hello";`

The reason is the prefix `L""` is for `wchar_t`, and the type `UChar` now means `char16_t` on RS3.

I don't know if you will run into any issues with this, but one method to work around this
```c++
#define UCHAR_TYPE wchar_t
#include "icu.h"
```

Reply:
We usually work with our own char16 type and then cast that to UChar whenever we need to, so that isn't an issue for us. Also our code builds using ICU 55 through 60 on various different platforms in various different circumstances so we should work with both. Regardless, it seems like icu.h is the one that will be updated going forward, so it seems like the better option.

Reply:
Yes, the main header to use going forward is `icu.h`.
I'm glad that you are using your own char16 type internally, and not using some platform specific type like wchar_t. 👍
(This will also help in terms of the cross-platform needs as well. )

Reply:
Yes, `char16` (and the related macro `_u("string literal"`) we use everywhere) were for the xplat source compat string typing concerns. The macro uses token pasting to create the appropriate string literal type depending on the platform.

Reply:
Ah neat! :)

Note: You might want to make sure you are setting the encoding as well on your source files as well then. Or at least for MSVC, not sure if other platforms just assume UTF-8 by default. 
`/source-charset:utf-8`

Reply:
We have a soft requirement (to be enforced by #4672) that all source files are ASCII to avoid any confusion -- but that flag would help too.